src/core/jwk.rs

```diff
@@ -402,6 +437,120 @@ impl
     }
 }
 
+///
+/// Variants of EdDSA Key.
+///
+pub enum CoreEdDsaKeyVariant {
+    ///
+    /// Ed25519 Variant
+    /// 
+    Ed25519,
+}
+
+struct EdDsaSigningKey {
+    variant: CoreEdDsaKeyVariant,
+    ed25519: Option<ed25519_dalek::SigningKey>,
+}
+
+impl EdDsaSigningKey {
+    fn from_pem(pem: &str, variant: CoreEdDsaKeyVariant) -> Result<Self, String> {
+        match variant {
+            CoreEdDsaKeyVariant::Ed25519 => Ok(Self {
+                variant,
+                ed25519: Some(ed25519_dalek::SigningKey::from_pkcs8_pem(pem).map_err(|err| err.to_string())?),
+            })
+        }
+    }
+
+    fn public_part(&self) -> Vec<u8> {
```

Comment:
I think I would remove this function and just put the `match` in `as_verification_key()`, since that function needs to know both the curve type and the public key